kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ParquetSchemaUtils.java

```diff
@@ -186,6 +186,43 @@ private static Type prunedType(Type type, DataType deltaType) {
       GroupType groupType = (GroupType) type;
       StructType structType = (StructType) deltaType;
       return groupType.withNewFields(pruneFields(groupType, structType));
+    } else if (type instanceof GroupType && deltaType instanceof ArrayType) {
+      // Handle array types with Parquet's 3-level structure:
+      // optional group array_field (LIST) {
+      //   repeated group list {
+      //     optional <element_type> element;
+      //   }
+      // }
+      GroupType arrayGroupType = (GroupType) type;
+      ArrayType arrayType = (ArrayType) deltaType;
+
+      // Assert proper 3-level array structure
+      checkArgument(
+          arrayGroupType.getFieldCount() == 1,
+          "In Parquet's 3-level structure, group type " + "must only contain one sub field");
+
+      Type listField = arrayGroupType.getType(0);
+      checkArgument(
+          listField.isRepetition(Type.Repetition.REPEATED),
+          "Array list field must be repeated: %s",
+          listField);
+
+      if (!(listField instanceof GroupType)) {
+        // For primitive arrays like array<int>, no pruning needed.
+        return type;
+      }
+
+      // array_field.list.element
+      GroupType listGroup = (GroupType) listField;
+      checkArgument(
+          listGroup.getFieldCount() == 1,
+          "Array list group must have exactly one element: %s",
+          listGroup);
+      Type elementType = listGroup.getType(0);
```

Comment:
Nit - maybe we can factor out a helper here + in [this code ](https://github.com/delta-io/delta/blob/3800ac40a0d3cccd97000862880ba5df67dd8e2d/kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/parquet/ArrayColumnReader.java#L57-L58) for fetching the parquet element type from a 3-level nested array type? Seems like thats the only real output we need here and I think would actually be more readable & combine shared code

Reply:
Done with the refactoring and add a TODO in map type